docs/DeveloperGuide.adoc

```diff
@@ -230,6 +230,119 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
+//end::SortCommand[]
+
+// tag::DeletTag[]
+=== DeletTag feature
+==== Current Implementation
+The delete tag command allows user to remove a particular tag from all persons in the AddressBook. For example, he may want to delete the tag "neigbour" from all his contacts if he is no longer staying at the place.
+
+We implement Model#deleteTag(Tag) torwards the direction of removeTag command, so that the developer dealing with the 'logic' component is able to implement such a command.
+
+==== Design Considerations
+
+===== Aspect: Implementation of time zone
+
+The current codebase has a flaw in tags management. Tags no longer in use by anyone may still exist on the AddressBook. This may cause some tests to fail.
+
+* **Alternative 1 (current choice):** After updating a person, we go through everyone in the address book and collect their tags.
+** Pros: No more unused tags.
+** Cons: The time complexity is too high, of O(n^2).
+* **Alternative 2:** When deleting a tag, check whether there are person still having it as a tag.
+** Pros: Less time consuming.
+** Cons: More complex inplementation.
+//end::DeletTag[]
+
+// tag::TotalPersonStatusBar[]
+=== TotalPersonStatusBar feature
+==== Current Implementation
+This feature will allow user to see how many contacts he or she has in his/her address book in a straight way, so that I can call it straightforward when I need the information.
+//end::TotalPersonStatusBar[]
+
+// tag::AddComment[]
+=== AddComment feature
+==== Current Implementation
+The add comment tag command  will allow user to add some plain text information associated with a contact that the user wants to remember, so that I can recall it when I need the information.
+
+To add the comment of a client, the cm/ tag is used with either the add or edit commands, followed by the comment.
+
+An example of the add command would be: add n/David ... cm/interested in football
+
+An example of the edit command would be: edit 1 cm/interested in bascketball
+
+==== Design Considerations
+
+===== Aspect: Implementation of comment
+
+* **Alternative 1 (current choice):** User must add comment when adding client.
+** Pros: Information consistency.
+** Cons: User may not always want to add comment to clients.
+* **Alternative 2:** User can optionally add the comments in a separate command.
+** Pros: Will use less memory (e.g. number of comments are less than or equal to number of users).
+** Cons: May be more difficult to manage contacts without some comments.
+//end::AddComment[]
+
+//tag::archive[]
+=== Archive feature
+==== Current Implementation
+The archive feature allows the salesperson to archive clients that he no longer contacts, but does not want to delete as he possible might want to contact him/her again. Once archived, the client is not shown in the list. To display the client, either find or listall command has to be used. The client is then displayed in a red font with strikethrough, indicating that he/she is archived.
```

Comment:
Please use the markdown's syntax for code using back ticks " \` \` ". E.g., command to search, find would look like `find` which is easier to read as a command keyword.